src/config/project/mod.rs

```diff
@@ -23,35 +19,144 @@ pub struct ProjectConfig {
+                continue;
+            };
+            // In the future this will be some more complex criteria
+            // like "closest package" or "has an oranda config"
+            // For now the criteria is "first one wins"
+            let is_better = best_workspace.is_none();
+            if is_better {
+                if let Some(defeated) = best_workspace {
+                    rejected_workspaces.push(defeated);
+                }
+                best_workspace = Some((workspace, pkg_idx));
+            } else {
+                rejected_workspaces.push((workspace, pkg_idx));
+            }
+        }
+
+        if let Some((best_ws, _best_pkg)) = &best_workspace {
+            // Report the winner
+            let message = format!("Detected {:?} project...", best_ws.kind);
+            Message::new(MessageType::Info, &message).print();
+            tracing::info!("{}", message);
+
+            // Warn about the existence of perfectly good losers
+            for (reject_ws, reject_pkg) in rejected_workspaces {
+                let reject_pkg = reject_ws.package(reject_pkg);
+                let message = format!(
+                    "Also found a {:?} project at {}, but we're ignoring it",
+                    reject_ws.kind, reject_pkg.manifest_path
+                );
+                Message::new(MessageType::Warning, &message).print();
+            }
         }
+
+        best_workspace
     }
-}
 
-#[derive(Debug, Eq, PartialEq)]
-pub enum Type {
-    Rust(Rust),
-    JavaScript(JavaScript),
+    /// Process the raw result of axoproject to print warnings and choose the actual
+    /// package in the workspace that applies.
+    fn handle_search_result(
+        start_dir: &Utf8Path,
+        search: WorkspaceSearch,
+        name: &str,
+    ) -> Option<(WorkspaceInfo, PackageIdx)> {
+        match search {
+            axoproject::WorkspaceSearch::Found(workspace) => {
+                // Now that we found the workspace, find the actual package that appears
+                // in the dir we're looking at.
+                let package = workspace.packages().find_map(|(idx, p)| {
+                    if p.manifest_path.parent().unwrap() == start_dir {
+                        Some(idx)
+                    } else {
+                        None
+                    }
+                });
+
+                if let Some(pkg_idx) = package {
+                    // Nice, this package is a perfect candidate
+                    Some((workspace, pkg_idx))
+                } else {
+                    // Found a workspace but none of the packages specifically control this dir.
+                    // This can happen if you run oranda in a dir with a virtual Cargo.toml.
+                    Message::new(
+                        MessageType::Warning,
+                        &format!("Ignoring {:?} project, none of the packages seem to own this directory", workspace.kind),
```

Comment:
the comment here makes more sense to me than the error message. i think it might be useful to mention that it's likely a cargo workspace and that oranda is per cargo "package" right now and so it should be run in the package dir not the workspace dir